src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/SetAzurePolicyDefinition.cs

```diff
@@ -160,7 +168,7 @@ private JToken GetResource(string resourceId, string apiVersion)
                     PolicyRule = string.IsNullOrEmpty(policyRuleJson) ? null : JObject.Parse(policyRuleJson),
```

Comment:
You could eliminate ```tempMode```, change ```existingMode``` from ```PolicyDefinitionMode?``` to ```PolicyDefinitionMode```, and initialize this way:
```
if (!Enum.TryParse<PolicyDefinitionMode>(resource.Properties["mode"]?.ToString(), true, out var existingMode))
{
    existingMode = PolicyDefinitionMode.None;
}
```
That allows you to just use ```existingMode``` here as your initializer:
```
   Mode = existingMode
```

Reply:
This would change what is sent in the request body (from null to NotSpecified) which I don't want to do